dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/basicToolApp/BasicTools.java

```diff
@@ -283,4 +289,33 @@ public City testObjectResponse(@ToolArg(name = "name", description = "name of yo
     }
 
     public record City(String name, String country, int population, boolean isCapital) {};
+
+    @Tool(name = "cancellationTool", title = "Cancellable tool", description = "A tool that waits to be cancelled")
+    public String cancellationTool(Cancellation cancellation) throws InterruptedException {
+        LOG.info("Cancelling Request");
+        int counter = 0;
+        TimeUnit.MILLISECONDS.sleep(500);
+        while (counter++ < 5) {
+            if (cancellation.check().isRequested()) {
+                LOG.info("Checking if tool is cancelled");
+                throw new OperationCancellationException();
+            }
+        }
```

Comment:
```suggestion
        while (counter++ < 20) {
            TimeUnit.MILLISECONDS.sleep(500);
            if (cancellation.check().isRequested()) {
                LOG.info("Checking if tool is cancelled");
                throw new OperationCancellationException();
            }
        }
```

- Sleep needs to inside the loop, but before the check
- Increase counter so we wait up to 10 seconds

Reply:
Done thanks